aea/helpers/async_utils.py

```diff
@@ -477,3 +458,206 @@ def __init__(self, getters: List[Tuple[Callable[[Any], None], Callable]]):
         super().__init__(
             [self._make_getter(handler, getter) for handler, getter in getters]
         )
+
+
+ready_future: Future = Future()
+ready_future.set_result(None)
+
+
+class Runnable(ABC):
+    """Abstract Runnable class."""
+
+    def __init__(self, loop=None, threaded=False) -> None:
```

Comment:
In `asyncio` APIs, the argument `loop` is going to be deprecated: 
https://stackoverflow.com/questions/60312374/what-are-all-these-deprecated-loop-parameters-in-asyncio

Of course, we are not writing an API for `asyncio`; however, considering removing the `loop` parameter would help in making this class neater. The same arguments apply, namely:
> - Prior to Python 3.6, asyncio.get_event_loop() was not guaranteed to return the event loop [...] This issue would be fixed in Python 3.6 and 3.5.3, where get_event_loop() was modified to reliably return the running loop if called from inside one, returning another_loop in the above scenario. Python 3.7 would additionally introduced get_running_loop() which completely ignores the global setting and always returns the currently running loop
> -  Once get_event_loop() became reliable, another problem was that of performance. Since the event loop was needed for some very frequently used calls, most notably call_soon, it was simply more efficient to pass around and cache the loop object. Asyncio itself did that, and many libraries followed suit. Eventually get_event_loop() was accelerated in C and was no longer a bottleneck.
> These two changes made the loop arguments redundant.
> [...] Just use get_event_loop() to get the loop when you need it. Alternatively, you can use get_running_loop() to assert that a loop is running.

Reply:
Not clear to me whether this means all the benefits (including performance issue) are present since 3.6 inclusive. We want to be fully supportive of 3.6

Reply:
of course, the change can be tackled in another PR.

Reply:
@DavidMinarsch it seems the fix is included in 3.7+: https://bugs.python.org/issue32296

Reply:
So this suggests to me we cannot remove the loop argument yet...